edit/cms_functions.php

```diff
@@ -0,0 +1,478 @@
+    $stmt = $db->prepare('INSERT INTO ' . $table. '(`value`) VALUES(:value)');
+    $stmt->bindParam(':value',$value);
+    return $stmt->execute();
+}
+
+/*
+ * Edits an existing single value entry in the given table
+ *
+ * @param int $id The ID of the entry to edit.
+ *
+ * @param string $table The table we are updating.
+ *
+ * @param string $value The value we are setting the row to.
+ *
+ * @return Returns TRUE if the project was edited successfully, FALSE if otherwise.
+ */
+function updateSingleValueInTable(int $id, string $table, string $value) {
+    if(!validateSingleTableRequest($table)) {
+        return FALSE;
+    }
+    $stmt = getPDO()->prepare('UPDATE ' . $table .  ' SET `value`=:value WHERE `id`=:id;');
+    $stmt->bindParam(':value',filter_var($value, FILTER_SANITIZE_STRING));
+    $stmt->bindParam(':id',$id);
+    return $stmt->execute();
+}
+
+/*
+ * Gets the contact info in the database under the given id.
+ *
+ * @param int id The unique ID of the target project.
+ *
+ * @return array Returns an associative array of the project's name, type, description, link and image OR false if there was no
+ * project of that ID in the database.
+ */
+function getSingleValueFromDB(int $id, string $table) {
+    if(!validateSingleTableRequest($table)) {
+        return FALSE;
+    }
+    $stmt = getPDO()->prepare('SELECT `value` FROM ' . $table . ' WHERE `id` = :id;');
+    $stmt->setFetchMode(PDO::FETCH_ASSOC);
+    $stmt->bindParam(':id',$id);
+    $stmt->execute();
+    $entry = $stmt->fetch();
+    return $entry;
+}
+
+/*
+ * Checks if we are editing the correct single column table, since targeting other tables with spoofed POST data is a thing.
+ *
+ * @param string $table The table the user is trying to change.
+ *
+ * return bool Returns TRUE if the table can be accessed in this way, otherwise FALSE.
+ */
+function validateSingleTableRequest($table) {
+    if($table !== 'icons' && $table !== 'badges') {
+        return FALSE;
+    }
+    return TRUE;
+}
+
+/*
+ * Checks if we are editing a valid table.
+ *
+ * @param string $table The given table to validate.
+ *
+ * return bool Returns TRUE if $table is a valid table, otherwise false.
+ */
+function validateListTableRequest(string $table) {
+    $values[0] = 'badges';
+    $values[1] = 'contact';
+    $values[2] = 'icons';
+    $values[3] = 'projects';
+    foreach ($values as $value) {
+        if($table === $value) {
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+/*
+ * Goes through each field, determining if it is empty, and returning true if it is. The normal empty function will merrily
+ * say that a comprising of '     ' is not empty, though it effectively appears as such.
+ *
+ * @param array $array The array of fields to check.
+ *
+ * @return Returns TRUE if the field meets our definition of empty, otherwise FALSE.
+ */
+function anyFieldEmpty(array $array) {
+    foreach ($array as $entry) {
+        $test = str_replace(' ','', $entry);
```

Comment:
http://php.net/manual/en/function.trim.php

Reply:
fixed in bcb8278

Is there a significant difference between the two?